File: common/src/main/java/com/ssblur/scriptor/word/action/potions/PhasingAction.java
package com.ssblur.scriptor.word.action.potions;

import com.ssblur.scriptor.api.word.Descriptor;
import com.ssblur.scriptor.blockentity.PhasedBlockBlockEntity;
import com.ssblur.scriptor.effect.ScriptorEffects;
import com.ssblur.scriptor.helpers.targetable.Targetable;

public class PhasingAction extends PotionAction {
  public PhasingAction() {
    super(ScriptorEffects.get(ScriptorEffects.PHASING), 20, 1d, new Cost(8, COSTTYPE.ADDITIVE));
  }

  @Override
  public void applyToPosition(Targetable caster, Targetable targetable, Descriptor[] descriptors, double strength, double duration) {
    var level = targetable.getLevel();
    var pos = targetable.getOffsetBlockPos();

    PhasedBlockBlockEntity.phase(level, pos, (int) Math.floor(duration * 3));
  }
}

File: common/src/main/kotlin/com/ssblur/scriptor/word/action/potions/PhasingAction.kt
package com.ssblur.scriptor.word.action.potions

import com.ssblur.scriptor.api.word.Descriptor
import com.ssblur.scriptor.blockentity.PhasedBlockBlockEntity.Companion.phase
import com.ssblur.scriptor.effect.ScriptorEffects.PHASING
import com.ssblur.scriptor.helpers.targetable.Targetable
import kotlin.math.floor

class PhasingAction : PotionAction(PHASING.ref(), 20.0, 1.0, Cost(8.0, COSTTYPE.ADDITIVE)) {
    override fun applyToPosition(
        caster: Targetable?,
        targetable: Targetable?,
        descriptors: Array<Descriptor>?,
        strength: Double,
        duration: Double
    ) {
        val level = targetable!!.level
        val pos = targetable.offsetBlockPos

        phase(level, pos, floor(duration * 3).toInt())
    }
}
